haystack/nodes/connector/crawler.py

```diff
@@ -236,18 +258,32 @@ def _write_to_files(
             else:
                 text = el.text
 
-            link_split_values = link.replace("https://", "").split("/")
-            file_name = f"{'_'.join(link_split_values)}.json"
-            file_path = output_dir / file_name
+            DocumentDict = Dict[str, Any]
 
-            data = {}
+            data: DocumentDict = {}
             data["meta"] = {"url": link}
             if base_url:
                 data["meta"]["base_url"] = base_url
             data["content"] = text
             document = Document.from_dict(data, id_hash_keys=id_hash_keys)
-            with open(file_path, "w", encoding="utf-8") as f:
-                json.dump(document.to_dict(), f)
+
+            if crawler_naming_function is not None:
+                file_name_preffix = crawler_naming_function(link, text)
+            else:
+                file_name_link = re.sub("[<>:'/\\|?*\0 ]", "_", link[:129])
+                file_name_hash = hashlib.md5(f"{link}".encode("utf-8")).hexdigest()
+                file_name_preffix = f"{file_name_link}_{file_name_hash[-6:]}"
+
+            file_path = output_dir / f"{file_name_preffix}.json"
```

Comment:
Minor, there's a typo (`preffix` should be `prefix`)
```suggestion
                file_name_prefix = f"{file_name_link}_{file_name_hash[-6:]}"

            file_path = output_dir / f"{file_name_prefix}.json"
```

Reply:
Oh, indeed! Just _affix_ uses double f's. :sweat_smile: